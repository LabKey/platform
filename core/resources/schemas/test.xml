<?xml version="1.0" encoding="UTF-8"?>
<!-- 

 * Copyright (c) 2004-2017 Fred Hutchinson Cancer Research Center
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.

 -->
<tables xsi:schemaLocation="http://labkey.org/data/xml ..\..\..\..\..\schemas\tableInfo.xsd" xmlns="http://labkey.org/data/xml" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<table tableName="TestTable" tableDbType="TABLE">
		<columns>
			<column columnName="_ts"/>
			<column columnName="EntityId"/>
			<column columnName="RowId"/>
			<column columnName="CreatedBy"/>
			<column columnName="Created"/>
			<column columnName="Container"/>
			<column columnName="Text"/>
			<column columnName="IntNull"/>
			<column columnName="IntNotNull"/>
			<column columnName="DatetimeNull"/>
			<column columnName="DatetimeNotNull"/>
			<column columnName="RealNull"/>
			<column columnName="BitNull"/>
			<column columnName="BitNotNull"/>
		</columns>
	</table>
    <table tableName="TestTable2" tableDbType="TABLE">
        <columns>
            <column columnName="_ts"/>
            <column columnName="EntityId"/>
            <column columnName="RowId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="Container"/>
            <column columnName="Text"/>
            <column columnName="IntNull"/>
            <column columnName="IntNotNull"/>
            <column columnName="DatetimeNull"/>
            <column columnName="DatetimeNotNull"/>
            <column columnName="RealNull"/>
            <column columnName="BitNull"/>
            <column columnName="BitNotNull"/>
        </columns>
    </table>
    <table tableName="TestTable3" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="_ts"/>
            <column columnName="EntityId"/>
            <column columnName="RowId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="Container"/>
            <column columnName="Text"/>
            <column columnName="IntNull"/>
            <column columnName="IntNotNull"/>
            <column columnName="DatetimeNull"/>
            <column columnName="DatetimeNotNull"/>
            <column columnName="RealNull"/>
            <column columnName="BitNull"/>
            <column columnName="BitNotNull"/>
        </columns>
    </table>
    <table tableName="Containers2" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="_ts"/>
            <column columnName="RowId"/>
            <column columnName="EntityId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="Parent"/>
            <column columnName="Name"/>
            <column columnName="SortOrder"/>
            <column columnName="Description"/>
            <column columnName="Type"/>
            <column columnName="Title"/>
            <column columnName="Searchable"/>
        </columns>
    </table>
    <table tableName="ContainerAliases2" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="Path"/>
            <column columnName="ContainerId"/>
        </columns>
    </table>
    <table tableName="Users2" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="Email"/>
            <column columnName="_ts"/>
            <column columnName="EntityId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="ModifiedBy"/>
            <column columnName="Modified"/>
            <column columnName="Owner"/>
            <column columnName="UserId"/>
            <column columnName="DisplayName"/>
            <column columnName="FirstName"/>
            <column columnName="LastName"/>
            <column columnName="Phone"/>
            <column columnName="Mobile"/>
            <column columnName="Pager"/>
            <column columnName="IM"/>
            <column columnName="Description"/>
            <column columnName="LastLogin"/>
            <column columnName="Active"/>
            <column columnName="HasPassword"/>
            <column columnName="ExpirationDate"/>
        </columns>
    </table>
</tables>
