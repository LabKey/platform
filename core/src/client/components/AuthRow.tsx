import React, { PureComponent } from 'react';
import { Col, Modal, Button } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
    faPencilAlt,
    faInfoCircle,
    faTimesCircle,
    faGripVertical,
    faCircle
} from '@fortawesome/free-solid-svg-icons';

import DynamicConfigurationModal from './DynamicConfigurationModal';
import DatabaseConfigurationModal from './DatabaseConfigurationModal';
import { AuthConfig, AuthConfigProvider } from "./models";

interface Props{
    authConfig: AuthConfig;
    index?: string;
    modalType?: AuthConfigProvider;
    configType?: string;
    canEdit: boolean;
    draggable: boolean;
    onDelete?: Function;
    toggleModalOpen?: Function;
    updateAuthRowsAfterSave?: Function;
}

interface State {
    editModalOpen: boolean;
    deleteModalOpen: boolean;
}

export default class AuthRow extends PureComponent<Props, Partial<State>> {
    constructor(props) {
        super(props);
        this.state = {
            editModalOpen: false,
            deleteModalOpen: false,
        };
    }

    onToggleModal = (localModalType: string, modalOpen: boolean): void => {
        this.setState(() => ({
            [localModalType]: !modalOpen,
        }));
    };

    render() {
        const {
            authConfig,
            modalType,
            configType,
            canEdit,
            draggable,
            toggleModalOpen,
            updateAuthRowsAfterSave,
            onDelete
        } = this.props;
        const isDatabaseAuth = authConfig.provider == 'Database';

        const handle = draggable && canEdit ? <LightupHandle /> : null;

        const enabledField = authConfig.enabled ? (
            <>
                <FontAwesomeIcon icon={faCircle} color="#75B666" /> &nbsp; Enabled
            </>
        ) : (
            <>
                <FontAwesomeIcon icon={faCircle} color="#999999" /> &nbsp; Disabled
            </>
        );

        const deleteIcon =
            canEdit && !isDatabaseAuth ? (
                <div className="clickable deleteIcon" onClick={() => this.setState({deleteModalOpen: true})}>
                    <FontAwesomeIcon icon={faTimesCircle} />
                </div>
            ) : null;

        const editOrViewIcon = canEdit ? (
            <div
                className="clickable editOrViewIcon"
                onClick={() => {
                    this.onToggleModal("editModalOpen", this.state.editModalOpen);
                    toggleModalOpen(true);
                }}>
                <FontAwesomeIcon size="1x" icon={faPencilAlt} />
            </div>
        ) : (
            <div className="clickable" onClick={() => this.onToggleModal("editModalOpen", this.state.editModalOpen)}>
                <FontAwesomeIcon size="1x" icon={faInfoCircle} />
            </div>
        );

        let modal;
        if (isDatabaseAuth) {
            modal =
                <DatabaseConfigurationModal
                    closeModal={() => {
                        this.onToggleModal("editModalOpen", this.state.editModalOpen);
                        toggleModalOpen(false);
                    }}
                    canEdit={canEdit}
                />
        } else {

            modal =
                <DynamicConfigurationModal
                    authConfig={authConfig}
                    configType={configType}
                    modalType={modalType}
                    canEdit={canEdit}
                    closeModal={() => {
                        this.onToggleModal("editModalOpen", this.state.editModalOpen);
                        if (canEdit) {
                            toggleModalOpen(false);
                        }
                    }}
                    updateAuthRowsAfterSave={updateAuthRowsAfterSave}
                />
        }

        const deleteModal =
            <Modal show={true} onHide={() => this.onToggleModal("deleteModalOpen", this.state.deleteModalOpen)}>
                <Modal.Header closeButton>
                    <Modal.Title>
                        Warning
                    </Modal.Title>
                </Modal.Header>
                <div className={"auth-row__delete-modal"}>
                    <div>
                        {`Are you sure you want to delete authentication configuration ${authConfig.description}?`}
                    </div>

                    <Button
                        className="labkey-button primary auth-row__confirm-delete"
                        onClick={onDelete}
                    >
                        Yes
                    </Button>
                </div>
            </Modal>;

        return (
            <div className="row-container">
                <div className="auth-row">
                    <div className="domain-row-container">
                        <div className="domain-row-handle">{handle}</div>

                        <div className="domain-row-main">
                            <Col xs={9} className="domain-row-base-fields">
                                <Col xs={4} className="description auth-row__field">
                                    {authConfig.description}
                                </Col>
                                <Col xs={4} className="details auth-row__field">
                                    {authConfig.details}
                                </Col>
                                <Col xs={3} className="provider auth-row__field">
                                    {authConfig.provider}
                                </Col>
                            </Col>

                            <Col xs={1} />

                            <Col xs={2} className="domain-row-base-fields">
                                <Col xs={7} className="enabled auth-row__field">
                                    {enabledField}
                                </Col>

                                <Col xs={1} className="delete auth-row__field">
                                    {deleteIcon}
                                </Col>

                                <Col xs={3} className="editOrView auth-row__field">
                                    {editOrViewIcon}
                                </Col>
                            </Col>

                            {this.state.editModalOpen && modal}
                            {this.state.deleteModalOpen && deleteModal}

                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

class LightupHandle extends PureComponent {
    render() {
        return (
            <div>
                <FontAwesomeIcon
                    size="lg"
                    icon={faGripVertical}
                />
            </div>
        );
    }
}
