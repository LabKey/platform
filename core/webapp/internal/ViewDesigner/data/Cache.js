/*
 * Copyright (c) 2015-2017 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
 */
Ext4.define('LABKEY.internal.ViewDesigner.QueryDetailsCache', {

    singleton: true,

    mixins : {
        observable: 'Ext.util.Observable'
    },

    constructor : function() {
        this.cache = {};
        this.results = {};

        this.mixins.observable.constructor.apply(this, arguments);

        this.addEvents('beforecacheresponse');
    },

    callback : function(callback, scope, result) {
        if (Ext4.isFunction(callback)) {
            callback.call(scope, result);
        }
    },

    add : function(keyConfig, result) {
        this.populateCache(this.getKey(keyConfig), result);
    },

    getKey : function(config) {

        if (!Ext4.isString(config.schema) || !Ext4.isString(config.query) ||
                Ext4.isEmpty(config.schema) || Ext4.isEmpty(config.query)) {
            throw 'Invalid cache key. Require a schema and a query.';
        }

        return JSON.stringify({
            schema: config.schema,
            query: config.query,
            view: !Ext4.isEmpty(config.view) ? config.view: undefined,
            fk: Ext4.isString(config.fk) ? config.fk : undefined
        });
    },

    inCache : function(key) {
        return this.cache[key] === true;
    },

    inFlightCache : function(key) {
        return Ext4.isArray(this.cache[key]);
    },

    loadCache : function(key, callback, scope) {
        // The value as an array denotes the cache resource is in flight
        if (!this.cache[key]) {
            this.cache[key] = [];
        }

        if (Ext4.isFunction(callback)) {
            this.cache[key].push({fn: callback, scope: scope});
        }
    },

    populateCache : function(key, result) {

        //console.log('# cols (start):', result.columns.length);
        this.fireEvent('beforecacheresponse', result);
        //console.log('# cols (end)  :', result.columns.length);

        this.results[key] = result;

        var cbs = this.cache[key];
        this.cache[key] = true;

        if (!Ext4.isEmpty(cbs)) {
            Ext4.each(cbs, function(config) {
                this.callback(config.fn, config.scope, result);
            }, this);
        }
    },

    getDetails : function(keyConfig, callback, scope) {
        var key = this.getKey(keyConfig);

        if (this.inCache(key)) {
            // -- cache hit
            this.callback(callback, scope, this.results[key]);
        }
        else if (this.inFlightCache(key)) {
            // -- cache miss, in flight
            this.loadCache(key, callback, scope);
        }
        else {
            // -- cache miss
            this.loadCache(key, callback, scope);

            // Request Details -- assume the current container
            LABKEY.Query.getQueryDetails({
                schemaName: keyConfig.schema,
                queryName: keyConfig.query,
                containerPath: keyConfig.containerPath,
                fk: keyConfig.fk,
                success: function(result) {
                    this.populateCache(key, result);
                },
                scope: this
            });
        }
    }
});

Ext4.define('LABKEY.internal.ViewDesigner.QueryDetailsProxy', {
    extend: 'Ext.data.proxy.Ajax',
    alias: 'proxy.querydetails',
    schema: undefined,
    query: undefined,
    view: undefined,
    url: '',
    read: function(operation, callback, scope) {
        var request = this.buildRequest(operation),
            params = {
                schema: this.schema,
                query: this.query,
                view: this.view,
                containerPath: this.containerPath
            },
            cacheCallback = this.createCacheCallback(request, operation, callback, scope);

        // get the paramters for this request
        if (!Ext4.isEmpty(request.params)) {
            // see if the 'node' parameter is in-use, could be a tree store
            if (!Ext4.isEmpty(request.params.node) &&
                    request.params.node !== LABKEY.internal.ViewDesigner.FieldMetaTreeStore.ROOT_ID) {
                if (operation.node && operation.node.get('fieldKey') !== undefined) {
                    params.fk = operation.node.get('fieldKey');
                }
            }
        }

        LABKEY.internal.ViewDesigner.QueryDetailsCache.getDetails(params, cacheCallback, this);
    },

    createCacheCallback : function(request, operation, callback, scope) {
        var me = this;
        return function(response) {
            me.processResponse(true, operation, request, response, callback, scope);
        }
    }
});