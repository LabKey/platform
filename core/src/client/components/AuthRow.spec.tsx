import React from 'react';

import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import { CAS_MODAL_TYPE } from '../../../test/data';

import AuthRow from './AuthRow';

describe('<AuthRow/>', () => {
    let component;

    beforeEach(() => {
        component = (
            <AuthRow
                description="My Configuration Name"
                details="Some Url"
                provider="CAS"
                enabled={true}
                draggable={true}
                modalType={CAS_MODAL_TYPE}
                canEdit={true}
            />
        );
    });

    test('Modal opens on click', () => {
        const wrapper = shallow<AuthRow>(component);
        const toggleModalOpen = jest.fn(() => {});
        wrapper.setProps({ toggleModalOpen });

        wrapper.instance().onToggleModal = jest.fn(() => wrapper.setState({ editModalOpen: true }));
        wrapper.update();

        expect(wrapper.state()).toHaveProperty('editModalOpen', false);

        const editIcon = wrapper.find('.clickable').last();

        editIcon.simulate('click');
        expect(wrapper.instance().onToggleModal).toHaveBeenCalled();

        expect(wrapper.state()).toHaveProperty('editModalOpen', true);
    });

    test('Non-draggable', () => {
        const wrapper = shallow<AuthRow>(component);
        wrapper.setProps({ draggable: false });

        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Editable', () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('View-only', () => {
        const wrapper = shallow<AuthRow>(component);
        wrapper.setProps({ canEdit: false });

        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
