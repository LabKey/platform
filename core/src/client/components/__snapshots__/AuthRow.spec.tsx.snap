// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthRow/> Editable 1`] = `
<div
  className="row-container"
>
  <div
    className="auth-row"
  >
    <div
      className="domain-row-container"
    >
      <div
        className="domain-row-handle"
      >
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-grip-vertical fa-w-10 fa-lg "
            data-icon="grip-vertical"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 32H32C14.33 32 0 46.33 0 64v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V64c0-17.67-14.33-32-32-32zm0 160H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm0 160H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zM288 32h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V64c0-17.67-14.33-32-32-32zm0 160h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm0 160h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="domain-row-main"
      >
        <div
          className="domain-row-base-fields col-xs-9"
        >
          <div
            className="description auth-row__field col-xs-4"
          >
            My Configuration Name
          </div>
          <div
            className="details auth-row__field col-xs-4"
          >
            Some Url
          </div>
          <div
            className="provider auth-row__field col-xs-3"
          >
            CAS
          </div>
        </div>
        <div
          className="col-xs-1"
        />
        <div
          className="domain-row-base-fields col-xs-2"
        >
          <div
            className="enabled auth-row__field col-xs-7"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle fa-w-16 "
              color="#75B666"
              data-icon="circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
               Enabled
          </div>
          <div
            className="delete auth-row__field col-xs-1"
          >
            <div
              className="clickable deleteIcon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times-circle fa-w-16 "
                data-icon="times-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="editOrView auth-row__field col-xs-3"
          >
            <div
              className="clickable editOrViewIcon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-pencil-alt fa-w-16 fa-1x "
                data-icon="pencil-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AuthRow/> Non-draggable 1`] = `
<div
  className="row-container"
>
  <div
    className="auth-row"
  >
    <div
      className="domain-row-container"
    >
      <div
        className="domain-row-handle"
      >
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-grip-vertical fa-w-10 fa-lg "
            data-icon="grip-vertical"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 32H32C14.33 32 0 46.33 0 64v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V64c0-17.67-14.33-32-32-32zm0 160H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm0 160H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zM288 32h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V64c0-17.67-14.33-32-32-32zm0 160h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm0 160h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="domain-row-main"
      >
        <div
          className="domain-row-base-fields col-xs-9"
        >
          <div
            className="description auth-row__field col-xs-4"
          >
            My Configuration Name
          </div>
          <div
            className="details auth-row__field col-xs-4"
          >
            Some Url
          </div>
          <div
            className="provider auth-row__field col-xs-3"
          >
            CAS
          </div>
        </div>
        <div
          className="col-xs-1"
        />
        <div
          className="domain-row-base-fields col-xs-2"
        >
          <div
            className="enabled auth-row__field col-xs-7"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle fa-w-16 "
              color="#75B666"
              data-icon="circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
               Enabled
          </div>
          <div
            className="delete auth-row__field col-xs-1"
          >
            <div
              className="clickable deleteIcon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times-circle fa-w-16 "
                data-icon="times-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="editOrView auth-row__field col-xs-3"
          >
            <div
              className="clickable editOrViewIcon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-pencil-alt fa-w-16 fa-1x "
                data-icon="pencil-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AuthRow/> View-only 1`] = `
<div
  className="row-container"
>
  <div
    className="auth-row"
  >
    <div
      className="domain-row-container"
    >
      <div
        className="domain-row-handle"
      >
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-grip-vertical fa-w-10 fa-lg "
            data-icon="grip-vertical"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 32H32C14.33 32 0 46.33 0 64v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V64c0-17.67-14.33-32-32-32zm0 160H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm0 160H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zM288 32h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V64c0-17.67-14.33-32-32-32zm0 160h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm0 160h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="domain-row-main"
      >
        <div
          className="domain-row-base-fields col-xs-9"
        >
          <div
            className="description auth-row__field col-xs-4"
          >
            My Configuration Name
          </div>
          <div
            className="details auth-row__field col-xs-4"
          >
            Some Url
          </div>
          <div
            className="provider auth-row__field col-xs-3"
          >
            CAS
          </div>
        </div>
        <div
          className="col-xs-1"
        />
        <div
          className="domain-row-base-fields col-xs-2"
        >
          <div
            className="enabled auth-row__field col-xs-7"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle fa-w-16 "
              color="#75B666"
              data-icon="circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
               Enabled
          </div>
          <div
            className="delete auth-row__field col-xs-1"
          >
            <div
              className="clickable deleteIcon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times-circle fa-w-16 "
                data-icon="times-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="editOrView auth-row__field col-xs-3"
          >
            <div
              className="clickable editOrViewIcon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-pencil-alt fa-w-16 fa-1x "
                data-icon="pencil-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
